cider-velox/test/SubstraitTest/VeloxPlanFragmentSubstraitConverterTest.cpp

```diff
@@ -100,6 +100,122 @@ class VeloxPlanFragmentSubstraitConverterTest : public OperatorTestBase {
+                  .orderBy({"c0 ASC NULLS FIRST", "c1 ASC NULLS LAST"}, true)
+                  .planNode();
+  assertQuery(plan, "SELECT * FROM tmp ORDER BY c0 NULLS FIRST, c1 NULLS LAST");
+  assertPlanConversion(
+      plan, plan, "SELECT * FROM tmp ORDER BY c0 NULLS FIRST, c1 NULLS LAST");
+}
+
+TEST_F(VeloxPlanFragmentSubstraitConverterTest, Limit) {
+  auto vectors = makeVector(3, 4, 2);
+  createDuckDbTable(vectors);
+  auto plan = PlanBuilder().values(vectors).limit(0, 10, false).planNode();
+
+  assertQuery(plan, "SELECT * FROM tmp LIMIT 10");
+  assertPlanConversion(plan, plan, "SELECT * FROM tmp LIMIT 10");
+}
+
+TEST_F(VeloxPlanFragmentSubstraitConverterTest, LimitPartial) {
+  auto vectors = makeVector(3, 4, 2);
+  createDuckDbTable(vectors);
+  auto plan = PlanBuilder().values(vectors).limit(0, 10, true).planNode();
+
+  assertQuery(plan, "SELECT * FROM tmp LIMIT 10");
+
+  assertPlanConversion(plan, plan, "SELECT * FROM tmp LIMIT 10");
+}
+
+TEST_F(VeloxPlanFragmentSubstraitConverterTest, LimitOffest) {
+  auto vectors = makeVector(3, 4, 2);
+  createDuckDbTable(vectors);
+  auto plan = PlanBuilder().values(vectors).limit(5, 10, false).planNode();
+
+  assertQuery(plan, "SELECT * FROM tmp OFFSET 5 LIMIT 10");
+  assertPlanConversion(plan, plan, "SELECT * FROM tmp OFFSET 5 LIMIT 10");
+}
+
+TEST_F(VeloxPlanFragmentSubstraitConverterTest, topN) {
+  auto vectors = makeVector(3, 4, 2);
+  createDuckDbTable(vectors);
+  auto plan =
+      PlanBuilder().values(vectors).topN({"c0 NULLS FIRST"}, 10, false).planNode();
+  assertPlanConversion(plan, plan, "SELECT * FROM tmp ORDER BY c0 NULLS FIRST LIMIT 10");
+}
+
+TEST_F(VeloxPlanFragmentSubstraitConverterTest, topNPartial) {
+  auto vectors = makeVector(3, 4, 2);
+  createDuckDbTable(vectors);
+  auto plan = PlanBuilder().values(vectors).topN({"c0 NULLS FIRST"}, 10, true).planNode();
+
+  assertQuery(plan, "SELECT * FROM tmp ORDER BY c0 NULLS FIRST LIMIT 10");
+
+  assertPlanConversion(plan, plan, "SELECT * FROM tmp ORDER BY c0 NULLS FIRST LIMIT 10");
+}
+
+TEST_F(VeloxPlanFragmentSubstraitConverterTest, topNFilter) {
+  auto vectors = makeVector(3, 4, 2);
+  createDuckDbTable(vectors);
+  auto plan = PlanBuilder()
+                  .values(vectors)
+                  .filter("c0 > 15")
+                  .topN({"c0 NULLS FIRST"}, 10, true)
+                  .planNode();
+  assertQuery(plan, "SELECT * FROM tmp WHERE c0 > 15 ORDER BY c0 NULLS FIRST LIMIT 10");
+
+  assertPlanConversion(
+      plan,
+      plan->sources()[0],
+      "SELECT * FROM tmp WHERE c0 > 15 ORDER BY c0 NULLS FIRST LIMIT 10");
+}
+
+TEST_F(VeloxPlanFragmentSubstraitConverterTest, topNTwoKeys) {
+  auto vectors = makeVector(3, 4, 2);
+  createDuckDbTable(vectors);
+  auto plan = PlanBuilder()
+                  .values(vectors)
+                  .filter("c0 > 15")
+                  .topN({"c0 NULLS FIRST", "c1 DESC NULLS LAST"}, 10, true)
+                  .planNode();
+
+  assertQuery(plan,
```

Comment:
Now cider is not support order by/offset/limit, so why these query cases passed?